src/logic/map_objects/world/critter.cc

```diff
@@ -100,12 +101,52 @@ CritterDescr::CritterDescr(const std::string& init_descname,
                            const LuaTable& table,
                            const World& world)
    : BobDescr(init_descname, MapObjectType::CRITTER, MapObjectDescr::OwnerType::kWorld, table),
-     editor_category_(nullptr) {
+     editor_category_(nullptr),
+     size_(0),
+     carnivore_(false),
+     appetite_(0),
+     reproduction_rate_(0) {
 	assign_directional_animation(&walk_anims_, "walk");
 
 	add_attributes(
 	   table.get_table("attributes")->array_entries<std::string>(), std::set<uint32_t>());
 
+	size_ = table.get_int("size");
```

Comment:
You can set this directly in the constructor and only have the check here. Then size_ can be const. Same for reproduction rate etc.

Reply:
done